lib/facter/resolvers/solaris/networking.rb

```diff
@@ -25,9 +24,14 @@ def read_facts(fact_name)
                 obtain_info_for_interface(lifreq)
               end
 
-              @fact_list = { interfaces: @interfaces } unless @interfaces.empty?
               @fact_list[:primary_interface] = Facter::Util::Resolvers::Networking::PrimaryInterface.read_from_route
 
+              unless @interfaces.empty?
+                @fact_list = { interfaces: @interfaces }
+                @fact_list[:primary_interface] ||=
+                  Facter::Util::Resolvers::Networking::PrimaryInterface.find_in_interfaces(@interfaces)
```

Comment:
Something similiar should be implemented for AIX networking_resolver

Reply:
AIX uses one call to get both primary and all bindings, if that fails, interfaces won't be available to search for a primary.